### DISCUSSION TITLE
Effect.merge vs withTaskGroup

### DISCUSSION THREAD

**Discussion by mariob** [pointfreeco/swift-composable-architecture#2469] [Q&A] [answered]
Is there any difference of using one of these two for just kicking of some background work, for example:

Task 1 - Fetch data from network and store in database (and quit)
Task 2 - Observe database and 'send' state update (in for await loop)

**Comment by kamcma** [comment, marked as answer]
[Effect `merge` uses `withTaskGroup` under the hood](https://github.com/pointfreeco/swift-composable-architecture/blob/41f450da3b962cd2508d5f74f3c4e486e723683f/Sources/ComposableArchitecture/Effect.swift#L264C1-L277C1). So it's a little bit personal preference, or if you need a particular Effect or TaskGroup API. TCA does ship [a helper](https://pointfreeco.github.io/swift-composable-architecture/1.2.0/documentation/composablearchitecture/withtaskcancellation(id:cancelinflight:operation:)) to get some granular cancellation behavior in effect world without having to drop down to TaskGroups.

**Reply by mariob** [reply]
@kamcma Thanks for the links! Effect.merge will work fine for my use cases.
